test/BlockNumberClockMode.t.sol

```diff
@@ -5,11 +5,27 @@ import {Test} from "forge-std/Test.sol";
 import {BlockNumberClockMode} from "../src/auto-delegates/extensions/BlockNumberClockMode.sol";
 import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
 
+contract MockBlockNumberClockMode is BlockNumberClockMode {
```

Comment:
This should have it's own file in the `mocks/` sudir

Reply:
Done: [d68cb0a](https://github.com/withtally/stGOV/pull/87/commits/d68cb0a9647a26184916f7e1d48961acd073b5a8)